mbed_devices/mbed_devices.py

```diff
@@ -12,8 +10,13 @@
 
 def get_connected_devices() -> Iterable[Device]:
     """Returns Mbed Devices connected to host computer."""
-    with ThreadPoolExecutor() as executor:
-        return [device for device in executor.map(_build_device, detect_candidate_devices())]
+    devices = []
+    for candidate_device in detect_candidate_devices():
+        try:
+            devices.append(_build_device(candidate_device))
+        except NoTargetForCandidate:
+            pass
+    return devices
```

Comment:
I wonder if there can be a bit more feedback to the user? I know you were talking about a verbose mode earlier. It would be nice to explain that a device was detected but that there was no reference to it in the targets db.

Reply:
When we add verbose mode, it'd set log level appropriately and vomit out information as the code assembles data. I think this would be enough feedback for curious souls. Granted we'll need to revisit the log messages when we get to implementing verbose mode. :)

Reply:
I think if we go that route we should have these messages at `DEBUG` level. Users are likely to have all kinds of removable USB drives connected to their machine that they don't need to be warned about.

Reply:
Yeah that would be nice. It looks like now we convert `MissingProductCode` and `UnknownTarget` into  `NoTargetForCandidate` which means we are losing information, but logging each of them first would help solve that if someone was debugging. Happy to leave it for later.

Reply:
@rosiemiller2 you're right that we'l lose information right now. I'll add some logging. :)

Reply:
Fixed https://github.com/ARMmbed/mbed-devices/pull/40/commits/62ade3ce96cd99b2d2852d45cb483be1f831c83b